sqlcapture/capture.go

```diff
@@ -30,30 +30,35 @@ func (ps *PersistentState) Validate() error {
 	return nil
 }
 
-// StreamsInState returns the IDs of all streams in a particular
-// state, in sorted order for reproducibility.
-func (ps *PersistentState) StreamsInState(modes ...string) []string {
-	var streams []string
-	for id, tableState := range ps.Streams {
-		for _, mode := range modes {
-			if tableState.Mode == mode {
-				streams = append(streams, id)
-				break
-			}
+// BindingsInState returns all the bindings having a particular persisted state, in sorted order for
+// reproducibility.
+func (c *Capture) BindingsInState(modes ...BackfillMode) []*Binding {
+	var bindings []*Binding
+	for _, binding := range c.Bindings {
+		if slices.Contains(modes, BackfillMode(c.State.Streams[binding.StateKey].Mode)) {
+			bindings = append(bindings, binding)
 		}
 	}
-	sort.Strings(streams)
-	return streams
+	slices.SortFunc(bindings, func(a, b *Binding) int {
+		if a.StreamID < b.StreamID {
```

Comment:
The body of this comparison function could just be `return strings.Compare(a.StreamID, b.StreamID)`, I think?